src/os_csyslogd/parser.c

```diff
@@ -0,0 +1,243 @@
+         	       sourceCopy + rm[g].rm_so);
+
+		strcpy((workingKeyStore->match), sourceCopy + rm[g].rm_so);
+		//printf("DEBUG: match: %s\n", match);
+		
+		if (g == 1){
+		    if (regexIndex == 0){
+			strcpy(workingKeyStore->shost, workingKeyStore->match);
+		        if (DEBUG != 0) printf("DEBUG: match: %s\n", workingKeyStore->match);	
+		    }
+		    else if (regexIndex == 1){
+		        strcpy(workingKeyStore->src, workingKeyStore->match);
+		        if (DEBUG != 0) printf("DEBUG: match: %s\n", workingKeyStore->match);	
+		    }
+		    else if (regexIndex == 2){
+		        strcpy(workingKeyStore->src, workingKeyStore->match);
+		        if (DEBUG != 0) printf("DEBUG: match: %s\n", workingKeyStore->match);	
+		    }
+		    else if (regexIndex == 3){
+		        strcpy(workingKeyStore->location, workingKeyStore->match);
+		        if (DEBUG != 0) printf("DEBUG: match: %s\n", workingKeyStore->match);	
+		    }
+		    else if (regexIndex == 5){
+			strcpy(workingKeyStore->cn1, workingKeyStore->match);
+		        if (DEBUG != 0) printf("DEBUG: match: %s\n", workingKeyStore->match);	
+		    }
+		    else if (regexIndex == 6){
+			findMatch(7, field, (workingKeyStore->match), workingKeyStore);
+ 		    }
+		    else if (regexIndex == 7){
+			strcpy(workingKeyStore->duser, workingKeyStore->match);
+		        if (DEBUG != 0) printf("DEBUG: match: %s\n", workingKeyStore->match);	
+		    }
+		    else if (regexIndex == 8){
+			findMatch(9, field, (workingKeyStore->match), workingKeyStore);
+		    }
+		    else if (regexIndex == 9){
+			strcpy(workingKeyStore->destinationDnsDomain, workingKeyStore->match);
+		        if (DEBUG != 0) printf("DEBUG: match: %s\n", workingKeyStore->match);	
+		    }
+		    else if (regexIndex == 10){
+			findMatch(11, field, (workingKeyStore->match), workingKeyStore);
+		    }
+		    else if (regexIndex == 11){
+			strcpy(workingKeyStore->duid, workingKeyStore->match);
+		        if (DEBUG != 0) printf("DEBUG: match: %s\n", workingKeyStore->match);	
+		    }
+		    else if (regexIndex == 12){
+			strcpy(workingKeyStore->externalId, workingKeyStore->match);
+		        if (DEBUG != 0) printf("DEBUG: match: %s\n", workingKeyStore->match);	
+		    }
+		    else if (regexIndex == 13){
+			strcpy(workingKeyStore->cat, workingKeyStore->match);
+		        if (DEBUG != 0) printf("DEBUG: match: %s\n", workingKeyStore->cat);	
+		    } 
+		     
+		}
+		else{
+		    if (regexIndex == 4){
+		        if (g == 2){
+			    findMatch(12, field, (workingKeyStore->match), workingKeyStore);
+		        }
+		        else if (g == 6){
+			    strcpy(workingKeyStore->sntdom, workingKeyStore->match); 
+		            if (DEBUG != 0) printf("DEBUG: match: %s\n", workingKeyStore->match);	
+		        }	
+		        else if (g == 7){
+			    strcpy(workingKeyStore->msg, workingKeyStore->match);
+		            if (DEBUG != 0) printf("DEBUG: match: %s\n", workingKeyStore->match);	
+		        }
+		    }
+		}
+		
+            }
```

Comment:
To avoid a memory leak, you should free the compiled expression at the end if it compiled successfully by adding:
``regfree(&re);``

Reply:
Thank you for catching this! I added some similar expression to the alert.c file but guess I missed doing it in the parser.c file. @vikman90 I will fix this bug and do a PR to the master branch. I would like to note these additions for CEF mappings are used for Windows events.

Reply:
Thanks @larriaga. 

You don't have to do a new PR, you only should change the base branch editing this PR.

Best regards.